test/e2e/cypress/integration/profiles/recovery/code/errors.spec.ts

```diff
@@ -0,0 +1,174 @@
+import { extractRecoveryCode, appPrefix, gen } from '../../../../helpers'
+import { routes as react } from '../../../../helpers/react'
+import { routes as express } from '../../../../helpers/express'
+
+context('Account Recovery Errors', () => {
+  ;[
+    {
+      recovery: react.recovery,
+      app: 'react' as 'react',
+      profile: 'spa'
+    },
+    {
+      recovery: express.recovery,
+      app: 'express' as 'express',
+      profile: 'recovery'
+    }
+  ].forEach(({ recovery, profile, app }) => {
+    describe(`for app ${app}`, () => {
+      before(() => {
+        cy.deleteMail()
+        cy.useConfigProfile(profile)
+        cy.proxy(app)
+      })
+
+      beforeEach(() => {
+        cy.deleteMail()
+        cy.longRecoveryLifespan()
+        cy.longCodeLifespan()
+        cy.disableVerification()
+        cy.enableRecovery()
+        cy.useRecoveryStrategy('code')
+      })
+
+      it('should invalidate flow if wrong code is submitted too often', () => {
+        const identity = gen.identityWithWebsite()
+        cy.registerApi(identity)
+        cy.visit(recovery)
+        cy.get(appPrefix(app) + "input[name='email']").type(identity.email)
+        cy.get("button[value='code']").click()
+        cy.get('[data-testid="ui/message/1060003"]').should(
+          'have.text',
+          'An email containing a recovery code has been sent to the email address you provided.'
+        )
+        for (let i = 0; i < 5; i++) {
+          cy.get("input[name='code']").type(i + '' + i) // Invalid code
+          cy.get("button[value='code']").click()
+          cy.get('[data-testid="ui/message/4060006"]').should(
+            'have.text',
+            'The recovery code is invalid or has already been used. Please try again.'
+          )
+          cy.noSession()
+        }
+
+        cy.get("input[name='code']").type('123123') // Invalid code
+        cy.get("button[value='code']").click()
+        cy.get('[data-testid="ui/message/4060007"]').should(
+          'have.text',
+          'The recovery was submitted too often. Please try again.'
+        )
+        cy.noSession()
```

Comment:
Please add one more assertion which enters the email again, to check if the form is properly reset and retriable :)

Reply:
Done.